src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -3650,25 +3650,24 @@ LogicalResult ONNXGatherNDOp::verify() {
         "<= " + std::to_string(dataRank - b));
 
   // All values in 'indices' are expected to satisfy the inequality:
-  //   -data.shape[i] <= indices[...,i] <= (data.shape[i]-1)].
-  for (int64_t i = 0; i < indicesRank; ++i) {
-    int64_t dataDimAtAxis = dataShape[i];
-    if (dataDimAtAxis < 0)
-      continue;
-
-    if (DenseElementsAttr valueAttribute =
-            getDenseElementAttributeFromONNXValue(indices))
-      for (IntegerAttr value : valueAttribute.getValues<IntegerAttr>()) {
-        static int n = 0;
-        int64_t index = value.getInt();
-        if (index < -dataDimAtAxis || index > dataDimAtAxis - 1)
+  //   -data.shape[b + i] <= indices[...,i] <= (data.shape[b + i]-1)].
```

Comment:
It makes sense. Gathering should start from `b+1` axis. 

So it looks like [the description](https://github.com/onnx/onnx/blob/main/docs/Operators.md#GatherND) in ONNX was wrong `All values in indices are expected to be within bounds [-s, s-1] along axis of size s (i.e.) -data_shape[i] <= indices[...,i] <= data_shape[i] - 1. It is an error if any of the index values are out of bounds.`, doesn't it?

Reply:
Yes I think the ONNX description is wrong unfortunately